File: moonphase/frame_calc.py
#!/usr/bin/env python3

from datetime import datetime, timezone

start = datetime(2026,1,1,tzinfo=timezone.utc)
now = datetime.now(timezone.utc)
hours = int((now - start).total_seconds() // 3600)
frame = hours + 1

print(frame)

# frame_str = f"{frame:04d}"

# url = (
#     "https://svs.gsfc.nasa.gov/vis/"
#     "a000000/a005500/a005587/"
#     "frames/1920x1080_16x9_30p/plain/"
#     f"moon.{frame_str}.tif"
# )

# print(url)

File: moonphase/frame_calc.lua
#!/usr/bin/env lua

local start = os.time({
    year = 2026,
    month = 1,
    day = 1,
    hour = 0,
    min = 0,
    sec = 0,
    isdst = false
})

local now = os.time()

local hours = math.floor((now - start) / 3600)
local frame = hours + 1

print(frame)